app/src/main/java/app/itaycsguy/musiciansaidb/MenuActivity.kt

```diff
@@ -106,16 +123,115 @@ class MenuActivity() : AppCompatActivity(), Parcelable {
+                    "The Image is automatically approved since you possess the right permission level"
+                }
+                Toast.makeText(this, infoText, Toast.LENGTH_LONG).show()
+            } else { Toast.makeText(this, "Pick an image prior to uploading operation!", Toast.LENGTH_LONG).show() }
+        }
+        findViewById<FloatingActionButton>(R.id.metaDataButton).setOnClickListener { _ ->
+            if(_imageHashPath == null) {
+                Toast.makeText(this, "Upload an image at first!", Toast.LENGTH_LONG).show()
             } else {
-                //TODO: change the user's permission to be an enum with the different permissions and address them all here.
-                "The Image is automatically approved since you possess the right permission level"
+                buildMetadataDialog()
             }
-            Toast.makeText(this, infoText, Toast.LENGTH_LONG).show()
         }
+    }
 
+    private fun buildMetadataDialog() : AlertDialog{
+        val currAct = this
+        val builder = AlertDialog.Builder(this)
+        builder.setTitle("Current-Image-Metadata")
+                .setView(layoutInflater.inflate(R.layout.activity_metadata,null))
+                .setPositiveButton("OK") { _, _ ->
+                    (_firebaseDB.getRef())?.let {
+                        val progressBar = startProgressBar(currAct, R.id.progressBar)
+                        try {
+                            _storageReference.child("${FirebaseDB.IMAGES_DB}/${FirebaseDB.TEMP_IMAGES}/$_imageHashPath")
+                            val writerName = _alertDialog.findViewById<EditText>(R.id.metadata_writer).text.toString()
+                            if (!writerName.isEmpty() && _placesLocation != null && _notesName != null) {
+                                val map = HashMap<String, String>()
+                                map["email"] = FirebaseDB.encodeUserEmail(_user.getEmail())
+                                map["writer"] = _alertDialog.findViewById<EditText>(R.id.metadata_writer).text.toString()
+                                map["note_name"] = _notesName.toString()
+                                map["location"] = _placesLocation.toString()
+                                map["upload_time"] = (System.currentTimeMillis()/1000).toString()
+                                try {
+                                    _firebaseDB.writeTempImagesMetadata(_imageHashPath.toString(), map)
+                                    Toast.makeText(currAct, "DB was updated successfully!", Toast.LENGTH_LONG).show()
+                                } catch (e: Exception) {
+                                    Toast.makeText(currAct, "Could not meet an updating", Toast.LENGTH_LONG).show()
+                                }
+                            } else {
+                                Toast.makeText(currAct, "All fields are required to be filled.", Toast.LENGTH_LONG).show()
```

Comment:
why are they all required? if he doesnt know who wrote it or where was it or w.e? do we want to really enforce them all?